xwiki-platform-core/xwiki-platform-sharepage/xwiki-platform-sharepage-ui/src/main/resources/XWiki/SharePage.xml

```diff
@@ -337,7 +337,7 @@ $!{escapetool.xml($message)}
 &lt;/pre&gt;
 #end
 
-&lt;p&gt;To visit this document online, open &lt;a href="${doc.getExternalURL()}"&gt;${doc.getExternalURL()}&lt;/a&gt; in your browser.&lt;/p&gt;
+&lt;p&gt;To visit this document online, open &lt;a href="${doc.getExternalURL()}"&gt;${util.decodeURI("${doc.getExternalURL()}")}&lt;/a&gt; in your browser.&lt;/p&gt;
```

Comment:
Note that if the problem is showing some technical content to users, it would be easy to replace the URL by some text. For example by replacing the text `To visit this document online, open <url> in your browser` by `You can <visit this document online> in your browser`.